src/components/Alert/Alert.tsx

```diff
@@ -28,48 +30,50 @@ export const Alert = (props: AlertProps) => {
-
     return (
-        <Card
-            style={style}
-            className={bAlert({corners}, spacing({py: 4, px: 5}, className))}
-            theme={theme}
-            view={view}
-            qa={qa}
-        >
-            <Flex gap="3" alignItems={align}>
-                {icon}
-                <Flex direction={layout === 'vertical' ? 'column' : 'row'} gap="5" grow>
-                    <Flex gap="2" grow className={bAlert('text-content')}>
-                        <Flex direction="column" gap="1" grow justifyContent={align}>
-                            {title}
-                            {message}
+        <AlertContextProvider layout={layout} view={view}>
+            <Card
+                style={style}
+                className={bAlert({corners}, spacing({py: 4, px: 5}, className))}
+                theme={theme}
+                view={view}
+                qa={qa}
+            >
+                <Flex gap="3" alignItems={align}>
+                    {props.icon || <Alert.Icon theme={theme} view={view} />}
+                    <Flex direction={layout === 'vertical' ? 'column' : 'row'} gap="5" grow>
+                        <Flex gap="2" grow className={bAlert('text-content')}>
+                            <Flex direction="column" gap="1" grow justifyContent={align}>
+                                {typeof props.title === 'string' ? (
+                                    <Alert.Title text={props.title} />
+                                ) : (
+                                    props.title
+                                )}
+                                {message}
+                            </Flex>
                         </Flex>
+                        {Array.isArray(props.actions) ? (
+                            <Alert.Actions items={props.actions} />
+                        ) : (
+                            props.actions
+                        )}
                     </Flex>
-                    {actions}
+                    {onClose && (
+                        <Button view="flat" onClick={onClose}>
+                            <Icon
+                                data={Xmark}
+                                size={DEFAULT_ICON_SIZE}
+                                className={colorText({color: 'secondary'})}
+                            />
+                        </Button>
+                    )}
                 </Flex>
-                {onClose && (
-                    <Button view="flat" onClick={onClose}>
-                        <Icon
-                            data={Xmark}
-                            size={DEFAULT_ICON_SIZE}
-                            className={colorText({color: 'secondary'})}
-                        />
-                    </Button>
-                )}
-            </Flex>
-        </Card>
+            </Card>
+        </AlertContextProvider>
     );
 };
 
 Alert.Icon = AlertIcon;
 Alert.Title = AlertTitle;
 Alert.Actions = AlertActions;
+Alert.Action = AlertAction;
```

Comment:
If it's a public API, then we should export types for these

Reply:
It is assumed that static component methods cannot be used in isolation from the component. So only `AlertProps` type is exported